pkg/network/usm/sharedlibraries/ebpf.go

```diff
@@ -50,27 +49,28 @@ var progSingleton *EbpfProgram
 // LibraryCallback defines the type of the callback function that will be called when a shared library event is detected
 type LibraryCallback func(LibPath)
 
-type libsetData struct {
-	libset      Libset
-	callbacks   map[*LibraryCallback]struct{}
-	done        chan struct{}
-	perfHandler *ddebpf.PerfHandler
-	enabled     bool
+type libsetHandler struct {
+	callbacksMutex sync.RWMutex
+	callbacks      map[*LibraryCallback]struct{}
+	done           chan struct{}
+	perfHandler    *ddebpf.PerfHandler
+	enabled        bool
+	requested      bool
```

Comment:
I'd add inline comments here
without reading the entire code, I'm not sure I can tell the difference between requested and enabled

Reply:
Added, let me know if something's missing.